lib/src/components/NxFileUpload/NxMultiFileUpload/NxMultiFileUpload.tsx

```diff
@@ -0,0 +1,188 @@
+      </span>
+      <NxTooltip title={buttonLabel}>
+        <button type="button" aria-label={buttonLabel} className="nx-selected-file__dismiss-btn" onClick={onDismiss}>
+          <NxFontAwesomeIcon icon={faTimesCircle} />
+        </button>
+      </NxTooltip>
+    </span>
+  );
+}
+
+const NxFileUpload = forwardRef<HTMLDivElement, Props>(function NxFileUpload(props, ref) {
+  const {
+        className: classNameProp,
+        onChange: onChangeProp,
+        files,
+        isRequired,
+        isPristine,
+        id,
+        disabled,
+        ...attrs
+      } = props,
+      isFileSelected = !!files?.length,
+      { showValidationErrors: formShowValidationErrors } = useContext(FormAriaContext),
+      showValidationErrors = formShowValidationErrors || !isPristine,
+      showError = isRequired && showValidationErrors && !isFileSelected,
+      className = classnames('nx-multi-file-upload', classNameProp),
+      noFileMessageClassName = classnames('nx-multi-file-upload__no-file-message', {
+        'nx-multi-file-upload__no-file-message--invalid': showError
+      }),
+      inputRef = useRef<HTMLInputElement>(null),
+      inputId = useUniqueId('nx-multi-file-upload-input', id),
+      selectedFilesContainerRef = useRef<HTMLDivElement>(null),
+      validationErrorId = useUniqueId('nx-multi-file-upload-validation-error'),
+      [previousFiles, setPreviousFiles] = useState<FileList | null>(files);
+
+  function onChange(evt: FormEvent<HTMLInputElement>) {
+
+    const dataTransferObject = new DataTransfer();
+
+    // get the previous files uploaded
+    if (files) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(files));
+    }
+
+    const inputFiles = evt.currentTarget.files,
+        normalizedFiles = !inputFiles?.length ? null : inputFiles;
+
+    if (normalizedFiles) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(normalizedFiles));
+    }
+
+    if (inputRef.current) {
+      inputRef.current.files = dataTransferObject.files;
+      onChangeProp(inputRef.current.files);
+    }
+  }
+
+  function openPicker() {
+    inputRef.current?.click();
+  }
+
+  function onDismiss(fileObj : File, selectedFile : HTMLElement | null) {
+    // if the selected file is either the last file in the container or the last remaining file, set focus accordingly
+    const closeBtns = Array.from(selectedFilesContainerRef.current?.
+        querySelectorAll<HTMLButtonElement>('.nx-selected-file__dismiss-btn') ?? []);
+    forEach((btn)=> {
+      if (selectedFile?.contains(btn) && btn === last(closeBtns)) {
+        closeBtns.length <= 1 ? inputRef.current?.focus() : closeBtns[closeBtns.length - 2].focus();
+      }
+    }, closeBtns);
+
+    const dataTransferObject = new DataTransfer();
+    if (inputRef.current) {
+      if (files) {
+        forEach((f: File) => dataTransferObject.items.add(f), without([fileObj], Array.from(files)));
+        inputRef.current.files = dataTransferObject.files;
```

Comment:
Similar to https://github.com/sonatype/sonatype-react-shared-components/pull/850/files#r1025274499, remove this

Reply:
>Similar to https://github.com/sonatype/sonatype-react-shared-components/pull/850/files#r1025274499, remove this

Removed here: [5d349a5](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/5d349a5cf8eb783e4203646b364fd3d5eac53d16)